pages/index.js

```diff
@@ -27,9 +32,11 @@ export default function HomePage() {
     (item) => !purchasedIds.includes(item._id)
   );
 
-  const purchasedItems = shoppingItems
-    .filter((item) => purchasedIds.includes(item._id))
-    .reverse();
+const purchasedItems = purchasedIds
+  .map(id => shoppingItems.find(item => item._id === id))
+  .filter(Boolean) // remove undefined if any
```

Comment:
First of all, why is this part of this PR? This is not part of the current scope of this featuree,,,

```suggestion
  .filter((item) => item !== undefined && item !== null)
```
Writing it out explicitly is better... This makes the intent crystal clear! 
You’re filtering out missing items, not just any falsy value.